treewide: mgmt: mcumgr: Change "ret" to "err" for SMP version 2

nordicjm opened the pull request:
This is a stable API treewide change changing the newly introduced "ret" response to "err" as it was overlooked that the shell_mgmt group already used "ret" to return the exit code of the command and this created a collision. Since SMP version 2 was only recently introduced, there should not be any public implementations of it as of yet.

Fixes #60980

Note: Not a treewide change as in touching lots of files, but a single commit making all needed changes so that it doesn't break a bisect

nordicjm committed:
treewide: mgmt: mcumgr: Change "ret" to "err" for SMP version 2

This is a stable API treewide change changing the newly introduced
"ret" response to "err" as it was overlooked that the shell_mgmt
group already used "ret" to return the exit code of the command
and this created a collision. Since SMP version 2 was only recently
introduced, there should not be any public implementations of it
as of yet, but the original function has been kept and marked as
deprecated.

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm commented:
@philips77 fixed, thanks

nashif commented:
Arch Review:

- No objections to change, deprecation is provided.
- No need for a backport to previous releases.

nordicjm committed:
doc: release: 3.5: Add MCUmgr SMP version 2 stable change note

Adds a note on the stable API change of changing the MCUmgr SMP
version 2 error entry name to avoid a collision with shell_mgmt
which already used "ret".

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>
